src/lib/Form/Type/Search/TrashSearchType.php

```diff
@@ -0,0 +1,97 @@
+use EzSystems\EzPlatformAdminUi\Form\Type\Section\SectionChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use eZ\Publish\API\Repository\PermissionResolver;
+
+class TrashSearchType extends AbstractType
+{
+    /** @var \Symfony\Contracts\Translation\TranslatorInterface */
+    private $translator;
+
+    /** @var \eZ\Publish\API\Repository\PermissionResolver */
+    private $permissionResolver;
+
+    /** @var \EzSystems\EzPlatformAdminUi\Form\Type\ChoiceList\Loader\TimePeriodChoiceLoader */
+    private $timePeriodChoiceLoader;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        PermissionResolver $permissionResolver,
+        TimePeriodChoiceLoader $timePeriodChoiceLoader
+    ) {
+        $this->translator = $translator;
+        $this->permissionResolver = $permissionResolver;
+        $this->timePeriodChoiceLoader = $timePeriodChoiceLoader;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('page', HiddenType::class)
+            ->add('content_type', ContentTypeChoiceType::class, [
+                'required' => false,
+                'placeholder' => /** @Desc("Any content types") */ 'trash.search.any_content_types',
+            ])
+            ->add('creator', UserType::class)
+            ->add('trashed_interval', DateIntervalType::class)
+            ->add('trashed', ChoiceType::class, [
+                'choice_loader' => $this->timePeriodChoiceLoader,
+                'required' => false,
+                'placeholder' => /** @Desc("Any time") */ 'trash.search.any_time',
+                'mapped' => false,
+            ])
+            ->add('sort', SortType::class)
+        ;
+
+        if ($this->permissionResolver->hasAccess('section', 'view') !== false) {
+            $builder->add('section', SectionChoiceType::class, [
+                'required' => false,
+                'multiple' => false,
+                'placeholder' => /** @Desc("Any section") */ 'trash.search.section.any',
+            ]);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \Symfony\Component\OptionsResolver\Exception\AccessException
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'data_class' => TrashSearchData::class,
+//            'error_mapping' => [
```

Comment:
rid of of this, or make it work if needed

Reply:
I think there will not be a form validation (because of every field is from "list")